src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -1030,6 +1030,7 @@ public function withCount($relations)
                 $relation->getRelated()->newQuery(), $this
             );
 
+            $this->mergeModelDefinedRelationWheresToHasQuery($query, $relation);
             call_user_func($constraints, $query);
```

Comment:
I think it should be very similar to this: https://github.com/laravel/framework/blob/5.2/tests/Database/DatabaseEloquentSoftDeletesIntegrationTest.php#L410:L427